### PR TITLE
DrivAerML: lower LR to 3e-4 at 4L/320d architecture

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The DrivAerML baseline uses lr=5e-4 with the 4L/320d architecture. For AirfRANS and TandemFoil, lower learning rates with cosine annealing improved convergence stability. The 4L/320d architecture has more parameters than previous configs, and a lower lr=3e-4 may allow more careful optimization of the loss landscape, reducing surface relative L2 error. This is especially relevant for the 3D car geometry where pressure gradients can be steep around complex body features.

## Instructions
Run the DrivAerML baseline with lr lowered to 3e-4:

```bash
SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 320 \
  --model-heads 5 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L320d-lr \
  --epochs 9999
```

Changes vs baseline:
- `--lr 3e-4` (was 5e-4)

## Baseline
Current best DrivAerML metrics (PR #2648, W&B: qx7z7if3):
- val_primary/surface_rel_l2_pct: **5.027%** (lower is better)

Reproduce baseline:
```bash
cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 320 --model-heads 5 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT levi:

## Results

### DrivAerML 4L/320d lr=3e-4 (W&B run aj4cqhdu)
- **val_primary/surface_rel_l2_pct:** 5.596% (final, epoch 266) — **best: 5.210% at epoch 262**
- **test_primary/surface_rel_l2_pct:** 6.842% (final)
- **Epochs:** 266 (181 min)
- **Peak memory:** GPU 0 (Blackwell RTX PRO 6000)
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py --dataset drivaerml --optimizer adamw --lr 3e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 320 --model-heads 5 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --epochs 9999 --agent levi --wandb-name "levi/drivaerml-4L320d-lr3e4" --wandb-group drivaerml-4L320d-lr`

### What happened
- **lr=3e-4 does NOT beat the lr=5e-4 baseline** (5.210% vs 5.027% = +3.6% worse).
- Convergence is very smooth and monotonic — steady descent from 63.09% to 5.21% over 266 epochs with no instability or divergence.
- The model was still improving at cutoff (best at epoch 262 of 266) — with more training time it might eventually reach 5.027%, but within 180 min it's clearly slower.
- Lower LR = more epochs (266 vs baseline's epoch count) but each epoch optimizes less aggressively, resulting in a net deficit.
- The cosine T_max=30 cycling doesn't cause instability at either LR, but lr=5e-4 reaches the optimal basin faster.

### Suggested follow-ups
- **lr=5e-4 remains optimal** for DrivAerML 4L/320d — no reason to lower it
- lr=3e-4 might match given 2x the training budget, but that's not practical
- If LR exploration is still of interest, lr=8e-4 (higher than baseline) might be worth testing — DrivAerML may benefit from more aggressive optimization

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2779 because ADVISOR: Closing as falsified hypothesis — clean negative result. W&B confirms val=5.210% vs 4.619% baseline, a 12.8% regression. This is the second independent falsification of the lower-LR hypothesis for DrivAerML (ray's #2797 at 4L/512d showed 6.9% regression). Two different architectures, same conclusion: lr=3e-4 hurts DrivAerML within the 180-min training budget. DrivAerML needs at least lr=5e-4 to make meaningful progress. Not worth pursuing further..
